src/config.c

```diff
@@ -2244,31 +2238,15 @@ static int updateTlsCfg(char *val, char *prev, const char **err) {
     }
     return 1;
 }
-static int updateTlsCfgBool(int val, int prev, const char **err) {
-    UNUSED(val);
-    UNUSED(prev);
-    return updateTlsCfg(NULL, NULL, err);
-}
-
-static int updateTlsCfgInt(long long val, long long prev, const char **err) {
-    UNUSED(val);
-    UNUSED(prev);
-    return updateTlsCfg(NULL, NULL, err);
-}
 
-static int updateTLSPort(long long val, long long prev, const char **err) {
-    /* Do nothing if port is unchanged */
-    if (val == prev) {
-        return 1;
-    }
-
-    /* Configure TLS if tls is enabled */
-    if (prev == 0 && tlsConfigure(&server.tls_ctx_config) == C_ERR) {
```

Comment:
I thought about it, the down side is that we don't waste time re-configuring tls (and loading cert files etc) when changing a port. This makes sense because a port change isn't really related to TLS configuration. But now we have a mechanism in the `.set` interface to return 2 when there's no change. So if there's no port change we won't call the apply function anyway. 

The idea was that for any place we use to have a quick exist in the update function because of no change we now have a generic mechanism to handle this and there's no need for the `prev` val in the apply function (for this use case) anymore.

Reply:
ok, but when changing the port we still do an unnecessary re-init of TLS, maybe this function should do some optimization on that? maybe not by checking the port, but by checking if we're already initialized?

Reply:
I'm not sure it's a big deal to re-init here but it definitely seems redundant in case we changed the port but TLS was already configured. I can easily fix this by adding a `int tlsIsConfigured()` function in _tls.c_ and do:
```c
if (!tlsIsConfigured() && tlsConfigure(&server.tls_ctx_config) == C_ERR) {
```
It seems that any other way will require accessing internal _tls.c_ stuff. This seems nicer to me than assuming TLS is configured based on the old port setting. @yossigo does this make sense? Can I add this new API to _tls.c_?

Reply:
@yoav-steinberg It makes sense. although I'm not sure we won't run into the need to supply `apply` with the `prev` config elsewhere.

Reply:
The basic design is that during the _set_ phase  you can check the _prev_ value. The _apply_ phase is there just to apply the current setting. This simplifies the implementation but brings a bit less flexibility - like forcing the apply functions to be idempotent but I feel this is actually a simplification which is worth it.
Anyway added `tlsIsConfigured`.